### PR TITLE
Tweak pre-creating empty outputs 🤢 to fix #8756.

### LINKED ISSUE #8756: Upload to data libraries from Galaxy filesystem via API is broken
But only when symlinking (i.e. passing `payload['link_data_only'] = 'link_to_files'`).

Affects only the `dev` branch (future release_20.01). According to `git bisect`, this was broken by PR https://github.com/galaxyproject/galaxy/pull/8625 (commit 8d78dc1 in particular).

Spotted by BioBlend cron test builds: https://travis-ci.org/galaxyproject/bioblend/jobs/588394255

There doesn't seem to be anything useful in the Galaxy logs:
```
galaxy.tools.actions.upload_common INFO 2019-09-28 00:57:17,354 tool upload1 created job id 1
galaxy.web_stack.handlers INFO 2019-09-28 00:57:17,401 Flushed transaction for Job[id=1,tool_id=upload1] (13.286 ms)
galaxy.web_stack.handlers INFO 2019-09-28 00:57:17,401 (Job[id=1,tool_id=upload1]) Handler 'main' assigned using 'db-self' assignment method
127.0.0.1 - - [28/Sep/2019:00:57:16 +0100] "POST /api/libraries/f2db41e1fa331b3e/contents?key=add0585fa960bd57eb16dd7025a7ab55 HTTP/1.1" 200 - "-" "python-requests/2.22.0"
galaxy.jobs.mapper DEBUG 2019-09-28 00:57:17,448 (1) Mapped job to destination id: local
galaxy.jobs.handler DEBUG 2019-09-28 00:57:17,490 (1) Dispatching to local runner
127.0.0.1 - - [28/Sep/2019:00:57:17 +0100] "GET /api/libraries/f2db41e1fa331b3e/contents/f2db41e1fa331b3e?key=add0585fa960bd57eb16dd7025a7ab55 HTTP/1.1" 200 - "-" "python-requests/2.22.0"
galaxy.jobs DEBUG 2019-09-28 00:57:17,513 (1) Persisting job destination (destination id: local)
galaxy.jobs DEBUG 2019-09-28 00:57:17,514 (1) Working directory for job is: /tmp/tmp.monC3Aiu5U/jobs_directory/000/1
galaxy.jobs.runners DEBUG 2019-09-28 00:57:17,521 Job [1] queued (30.936 ms)
galaxy.jobs.handler INFO 2019-09-28 00:57:17,535 (1) Job dispatched
galaxy.jobs.command_factory INFO 2019-09-28 00:57:17,650 Built script [/tmp/tmp.monC3Aiu5U/jobs_directory/000/1/tool_script.sh] for tool command [python '/usr/users/ga002/soranzon/software/galaxyproject_galaxy/test/functional/tools/upload.py' /usr/users/ga002/soranzon/software/galaxyproject_galaxy /tmp/tmp.monC3Aiu5U/jobs_directory/000/1/registry.xml /tmp/tmp.monC3Aiu5U/jobs_directory/000/1/upload_params.json 1:/tmp/tmp.monC3Aiu5U/jobs_directory/000/1/dataset_1_files:None 2:/tmp/tmp.monC3Aiu5U/jobs_directory/000/1/dataset_2_files:None]
galaxy.jobs.runners DEBUG 2019-09-28 00:57:17,683 (1) command is: rm -rf working outputs; mkdir -p working outputs; cd working; /bin/bash /tmp/tmp.monC3Aiu5U/jobs_directory/000/1/tool_script.sh > ../outputs/tool_stdout 2> ../outputs/tool_stderr; return_code=$?; cd '/tmp/tmp.monC3Aiu5U/jobs_directory/000/1'; 
[ "$GALAXY_VIRTUAL_ENV" = "None" ] && GALAXY_VIRTUAL_ENV="$_GALAXY_VIRTUAL_ENV"; _galaxy_setup_environment True
python "metadata/set.py"; sh -c "exit $return_code"
galaxy.jobs.runners.local DEBUG 2019-09-28 00:57:17,693 (1) executing job script: /tmp/tmp.monC3Aiu5U/jobs_directory/000/1/galaxy_1.sh
galaxy.jobs DEBUG 2019-09-28 00:57:17,701 (1) Persisting job destination (destination id: local)
galaxy.jobs.runners.util.process_groups DEBUG 2019-09-28 00:57:20,398 check_pg(): No process found in process group 16321
galaxy.jobs.runners.local DEBUG 2019-09-28 00:57:20,399 execution finished: /tmp/tmp.monC3Aiu5U/jobs_directory/000/1/galaxy_1.sh
galaxy.jobs DEBUG 2019-09-28 00:57:20,433 (1) setting dataset 1 state to ERROR
galaxy.jobs DEBUG 2019-09-28 00:57:20,443 (1) setting dataset 2 state to ERROR
galaxy.jobs INFO 2019-09-28 00:57:20,524 Collecting metrics for Job 1 in /tmp/tmp.monC3Aiu5U/jobs_directory/000/1
galaxy.jobs DEBUG 2019-09-28 00:57:20,541 job 1 ended (finish() executed in (129.652 ms))
```

### PR DESCRIPTION
**Pull request by jmchilton**
Also better logging for failed uploads to make debugging issues like #8756 more clear.

### CODE REVIEW

File: lib/galaxy/tool_util/provided_metadata.py [hunk 1]
```diff
@@ -53,6 +53,8 @@ def get_new_datasets(self, output_name):
 
     def has_failed_outputs(self):
         """Determine if generation of any of the outputs failed.
+
+        If True, this method should also log information about at the first such failed output.
```
Comment (martenson) [inline lib/galaxy/tool_util/provided_metadata.py]:
```suggestion
        If True, this method should also log information about such failed outputs.
```

Comment (jmchilton) [reply lib/galaxy/tool_util/provided_metadata.py]: I meant for this to be "at least the first such failed output." - do you still prefer what you have here? This doesn't log about each failed outputs with these changes, only the first.

Comment (martenson) [reply lib/galaxy/tool_util/provided_metadata.py]: Noe, yours is clear. I wasn't sure what you wanted here.

**Comment by jmchilton** [comment]
Test failure looks to be unrelated.

**Comment by martenson** [comment]
Thanks @jmchilton

**Comment by nsoranzo** [comment]
Thanks @jmchilton, BioBlend tests on Galaxy dev branch pass again!

**Pull request merged**